utils/pythonutils/poetryutils.go

```diff
@@ -0,0 +1,134 @@
+package pythonutils
+
+import (
+	"errors"
+	"os"
+
+	"github.com/BurntSushi/toml"
+)
+
+type PyprojectToml struct {
+	Tool map[string]PoetryPackage
+}
+type PoetryPackage struct {
+	Name            string
+	Version         string
+	Dependencies    map[string]interface{}
+	DevDependencies map[string]interface{} `toml:"dev-dependencies"`
+}
+
+type PoetryLock struct {
+	Package []*PoetryPackage
+}
+
+// Executes the poetry-dependency-map script and returns a dependency map of all the installed poetry packages in the current environment and another list of the top level dependencies
+func getPoetryDependencies(srcPath string) (graph map[string][]string, directDependencies []string, err error) {
+	filePath, err := getPoetryLockFilePath(srcPath)
+	if err != nil || filePath == "" {
+		// Error was returned or poetry.lock does not exist in directory.
+		return nil, nil, err
+	}
+	projectName, directDependencies, err := getPackageNameFromPyproject(srcPath)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Extract packages names from poetry.lock
+	dependencies, dependenciesVersions, err := extractPackagesFromPoetryLock(filePath)
+	if err != nil {
+		return nil, nil, err
+	}
+	graph = make(map[string][]string)
+	// Add the root node - the project itself.
+	for _, directDependency := range directDependencies {
+		directDependencyName := directDependency + ":" + dependenciesVersions[directDependency]
+		graph[projectName] = append(graph[projectName], directDependencyName)
+	}
+	// Add versions to all dependencies
+	for dependency, transitiveDependencies := range dependencies {
+		for _, transitiveDependency := range transitiveDependencies {
+			transitiveDependencyName := transitiveDependency + ":" + dependenciesVersions[transitiveDependency]
+			graph[dependency] = append(graph[dependency], transitiveDependencyName)
+		}
+	}
+	return graph, graph[projectName], nil
+}
+
+func getPackageNameFromPyproject(srcPath string) (string, []string, error) {
+	filePath, err := getPyprojectFilePath(srcPath)
+	if err != nil || filePath == "" {
+		// Error was returned or pyproject.toml does not exist in directory.
+		return "", nil, err
```

Comment:
It may be safer to return an empty array instead of nil.
The sane goes for the below return statement.